src/iidm/StaticVarCompensatorAdder.cpp

```diff
@@ -24,12 +24,15 @@ StaticVarCompensatorAdder::StaticVarCompensatorAdder(VoltageLevel& voltageLevel)
 StaticVarCompensator& StaticVarCompensatorAdder::add() {
     checkBmin(*this, m_bMin);
     checkBmax(*this, m_bMax);
+    checkRegulatingTerminal(*this, m_regulatingTerminal, getNetwork());
     checkSvcRegulator(*this, m_voltageSetpoint, m_reactivePowerSetpoint, m_regulationMode);
 
-    std::unique_ptr<StaticVarCompensator> ptrSvc = stdcxx::make_unique<StaticVarCompensator>(getNetwork(), checkAndGetUniqueId(), getName(), m_bMin, m_bMax, m_voltageSetpoint, m_reactivePowerSetpoint, *m_regulationMode);
+    std::unique_ptr<Terminal> terminalAdded = checkAndGetTerminal();
+    std::unique_ptr<StaticVarCompensator> ptrSvc = stdcxx::make_unique<StaticVarCompensator>(getNetwork(), checkAndGetUniqueId(), getName(), m_bMin, m_bMax, m_voltageSetpoint,
+        m_reactivePowerSetpoint, *m_regulationMode, m_regulatingTerminal ? m_regulatingTerminal.get() : *terminalAdded);
```

Comment:
Could you extract the code to initialize the regulatingTerminal  and store the result in a local variable:
```
    std::unique_ptr<Terminal> ptrTerminal = checkAndGetTerminal();
    Terminal& regulatingTerminal = m_regulatingTerminal ? m_regulatingTerminal.get() : *ptrTerminal;
    std::unique_ptr<StaticVarCompensator> ptrSvc = stdcxx::make_unique<StaticVarCompensator>(getNetwork(), checkAndGetUniqueId(), getName(), m_bMin, m_bMax, m_voltageSetpoint,
        m_reactivePowerSetpoint, *m_regulationMode, regulatingTerminal);
    ...
   Terminal& terminal = svc.addTerminal(std::move(ptrTerminal));
```